### PR TITLE
#163745944 Update Parties controllers to persist data

### PR DESCRIPTION
**Pull request by ibkadeeko**
#### What does this PR do?
Update parties controller to persist data

#### Description of Task to be completed?
Rewrite parties controller to query a database and persist data
Check that all endpoints on offices routes are working properly
write unit tests for all endpoints
`POST /api/v1/parties`
`GET /api/v1/parties`
`GET /api/v1/parties/:id`
`PATCH /api/v1/parties/:id`
`DELETE /api/v1/parties/:id`

#### How should this be manually tested?

- First clone the project on your local computer. 
- To do that create a folder in your local machine, open the terminal in that folder and run `https://github.com/ibkadeeko/Politico.git`
- cd into the project on your terminal and run `npm install` to install all dependencies
- start the server by running `npm start`

Using Postman test the endpoint above
- Make requests to the routes above with varying values
- The data should persist

-Using Mocha run `npm test`

#### What are the relevant pivotal tracker stories
#163745944

### CODE REVIEW

File: server/controllers/parties.js [hunk 1]
```diff
@@ -1,97 +1,118 @@
...
+      if (err) {
+        res.status(400).send({
+          status: 400,
+          error: err.stack,
+        });
+      }
+      if (response.rowCount !== 0) {
+        return res.status(200).json({
+          status: 200,
+          data: [{ message: `User Successfully deleted with ID: ${id}` }],
```
Comment (tutugodfrey) [inline server/controllers/parties.js]: You don't need to expose the id of the deleted user. The message `User Successfully deleted ` is enough.

File: server/controllers/parties.js [hunk 2]
```diff
@@ -1,97 +1,118 @@
...
-    const partyIndex = partiesDb.findIndex(party => party.id === id);
-
-    if (partyIndex === -1) {
-      return res.status(404).json({
+    const { name } = req.body;
+    db.query('UPDATE parties SET name = $1 WHERE id = $2 RETURNING *', [name, id], (err, updatedParty) => {
+      if (err) {
+        res.status(400).send({
+          status: 400,
+          error: err.stack,
```
Comment (tutugodfrey) [inline server/controllers/parties.js]: Is the error message is `err.stack` descriptive enough as feedback for users? Would you rather return a custom error message.

File: server/controllers/parties.js (hunk 1 above)
Comment (ibkadeeko) [reply server/controllers/parties.js]: Ok

File: server/controllers/parties.js (hunk 2 above)
Comment (ibkadeeko) [reply server/controllers/parties.js]: I'll change the error message

Comment (ibkadeeko) [reply server/controllers/parties.js]: I have implemented the feedback

**Review by tutugodfrey — APPROVED** [review]

**Pull request merged**